pkg/hsskey/filereader_test.go

```diff
@@ -0,0 +1,418 @@
+		{
+			name:   "zero bytes",
+			expect: false,
+			data:   make([]byte, 64),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := validateChecksum(tt.data)
+			if result != tt.expect {
+				t.Fatalf("validateChecksum(%v) = %v, want %v", tt.data, result, tt.expect)
+			}
+		})
+	}
+}
+
+func TestDeduplicate(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    [][]uint8
+		expected [][]uint8
+	}{
+		{
+			name: "No duplicates",
+			input: [][]uint8{
+				{1, 2, 3},
+				{4, 5, 6},
+				{7, 8, 9},
+			},
+			expected: [][]uint8{
+				{1, 2, 3},
+				{4, 5, 6},
+				{7, 8, 9},
+			},
+		},
+		{
+			name: "With duplicates",
+			input: [][]uint8{
+				{1, 2, 3},
+				{1, 2, 3},
+				{4, 5, 6},
+			},
+			expected: [][]uint8{
+				{1, 2, 3},
+				{4, 5, 6},
+			},
+		},
+		{
+			name: "All duplicates",
+			input: [][]uint8{
+				{1, 2, 3},
+				{1, 2, 3},
+				{1, 2, 3},
+			},
+			expected: [][]uint8{
+				{1, 2, 3},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := deduplicate(tt.input)
+			if !reflect.DeepEqual(result, tt.expected) {
+				t.Errorf("deduplicate(%v) = %v, want %v", tt.input, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestGetHssEepromPaths(t *testing.T) {
+	tests := []struct {
+		name          string
+		pattern       string
+		allFiles      []string
+		expectedFiles []string
+		isErr         bool
+	}{
+		{
+			name:    "Matching all EEPROMs",
+			pattern: "*",
+			allFiles: []string{
+				"/sys/bus/i2c/devices/0-0050/eeprom",
+				"/sys/bus/i2c/devices/1-0050/eeprom",
+				"/sys/bus/i2c/devices/1-0050/device",
+			},
+			expectedFiles: []string{
+				"/sys/bus/i2c/devices/0-0050/eeprom",
+				"/sys/bus/i2c/devices/1-0050/eeprom",
+			},
+			isErr: false,
+		},
+		{
+			name:    "Matching multiple EEPROMs",
+			pattern: "0-007[01]",
+			allFiles: []string{
+				"/sys/bus/i2c/devices/0-0070/eeprom",
+				"/sys/bus/i2c/devices/0-0071/eeprom",
+				"/sys/bus/i2c/devices/1-0070/device",
+			},
+			expectedFiles: []string{
+				"/sys/bus/i2c/devices/0-0070/eeprom",
+				"/sys/bus/i2c/devices/0-0071/eeprom",
+			},
+			isErr: false,
+		},
+		{
+			name:    "Matching multiple EEPROMs behind mux",
+			pattern: "0-0050/channel-[01]/*",
+			allFiles: []string{
+				"/sys/bus/i2c/devices/0-0050/channel-0/0-0070/eeprom",
+				"/sys/bus/i2c/devices/0-0050/channel-1/0-0075/eeprom",
+				"/sys/bus/i2c/devices/0-0055/eeprom",
+			},
+			expectedFiles: []string{
+				"/sys/bus/i2c/devices/0-0050/channel-0/0-0070/eeprom",
+				"/sys/bus/i2c/devices/0-0050/channel-1/0-0075/eeprom",
+			},
+			isErr: false,
+		},
+		{
+			name:    "No matching found",
+			pattern: "0-0050",
+			allFiles: []string{
+				"/sys/bus/i2c/devices/0-0070/eeprom",
+			},
+			expectedFiles: []string{},
+			isErr:         true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create temp dir
+			tempDir, err := os.MkdirTemp("", "")
```

Comment:
Consider using https://pkg.go.dev/testing#T.TempDir

so 

`tempDir := t.TempDir()` 

It handles errors, and removal for ya

Reply:
Done.